src/dir.rs

```diff
@@ -360,7 +360,15 @@ fn _rename_flags(old_dir: &Dir, old: &CStr, new_dir: &Dir, new: &CStr,
 impl AsRawFd for Dir {
     #[inline]
     fn as_raw_fd(&self) -> RawFd {
-        self.0
+        let fd = match self.0 {
+            libc::AT_FDCWD => unsafe { libc::openat(self.0, DOT.as_ptr(), 0) },
```

Comment:
This leaks file descriptor. Because as_raw_fd returns borrowed fd, so nobody will close it.

Reply:
Indeed, I missed that details of the trait.